### ISSUE TITLE
Add focused, synthetic, future-proof autoconf scripts test

### ISSUE THREAD

**Issue by jneira** [haskell/cabal#7762]
@mikolaj suggestion in irc and recorded in #7671

> add focused, synthetic, future-proof autoconf scripts test with many sample generated files and/or add tests of compiling newest versions of network, process, time, old-time and any other important autoconf-using packages

Not sure how feasible could be the setup for those tests, specially the first mentioned ones. Otoh network is being tested in the new dogfooding job (see https://github.com/haskell/cabal/issues/7742)

To consider.

**Comment by Mikolaj** [comment]
Yes, I agree, cheaper is better, because we lack resources. I think it would be even stronger (than focused tests on our side) if a couple of popular packages had CI jobs that:

* build both with new and old autoconf (later on, with whatever autoconf version with breaking changes is still in use) and 
* build both with pre-generated files and files generated as part of the test. 

Newest stable cabal for all 4 modes of test should probably be enough, because we don't plan to change this part and, if autoconf breaks again, we will again try to fix cabal ahead of the changes being picked up by normal packages.

**Comment by Mikolaj** [comment]
Some work has already been done in package network by @hasufell: https://github.com/haskell/network/pull/516

I don't know how much is missing.
